src/Generator/EntityGenerator.php

```diff
@@ -842,17 +746,11 @@ private function isMatchingConditions($relationConditions, $randomRelation)
      *
      * @param string $relationName
      * @param array $relationConditions
-     * @param string $entityName
      *
      * @return string
      */
-    private function getRandomRelationId($relationName, $relationConditions, $entityName)
+    private function getRandomRelationId($relationName, $relationConditions)
     {
-        // if the current relation is also inside an already generated relation, use it instead of a random value
-        if ($relation = $this->getRelationFromDependencies($relationName, $entityName)) {
```

Comment:
Careful it seems like you remove a feature that may be important here, if you create random relations but a matching one already exists you should use it I guess no? Or at least have an option that states if you need to create a new one at each time or not

But since you now have a constraint of unicity it means you won't be able to always create an alternative random relation so you'll have to find the existing one